index.bs

```diff
@@ -1566,74 +1566,72 @@ Observers involve defining a new class, dictionaries for options, and a new type
-
-[[!WEBIDL]] contains many numeric types. However, it is very
-rare that its more specific ones are actually appropriate.
+<h2 id="types-and-units">Web IDL, Types, and Units</h2>
 
-JavaScript has only one numeric type, Number: IEEE 754 double-precision
-floating point, including ±0, ±Infinity, and NaN (although thankfully only one). The Web IDL "types" are coercion rules that apply when accepting an argument or triggering a setter. For example, a Web IDL <code>unsigned short</code> roughly says: "when someone passes this as an argument, take it modulo 65536 before doing any further processing". That is very rarely a useful thing to do.
+<h3 id="numeric-types">Use numeric types appropriately</h3>
 
-Instead, you will want to stick with one of:
+If an API you're designing uses numbers,
+use one of the following [[!WEBIDL]] numeric types,
+unless there is a specific reason not to:
 
   : <code>unrestricted double</code>
-  :: When truly any JavaScript number will do, including infinities and NaN
+  :: Any JavaScript number, including infinities and NaN
 
   : <code>double</code>
-  :: Any JavaScript number excluding infinities and NaN
+  :: Any JavaScript number, excluding infinities and NaN
 
   : <code>[EnforceRange] long long</code>
-  :: Any JavaScript number in the integer-representable range, throwing a <code highlight="js">TypeError</code>
-      outside the range and rounding inside of it
+  :: Any JavaScript number between -2<sup>53</sup> and 2<sup>53</sup>,
+     rounded to the nearest integer.
+     If a number outside this range is given,
+     the generated bindings will throw a <code highlight="js">TypeError</code>.
 
   : <code>[EnforceRange] unsigned long long</code>
-  :: Any nonnegative JavaScript number in the integer-representable range, throwing a
-      <code highlight="js">TypeError</code> outside the range and rounding inside of it
-
-  : <code>bigint</code>
-  :: When the possible integer values range more broadly than JavaScript numbers can accurately represent
-
-Additionally, you can combine any of the above with an extra line in your
-algorithm to validate that the number is within the expected domain-specific
-range, and throwing or performing other actions in response. (While it is very
-rarely appropriate to modify author input by taking it modulo 65536, it might be
-appropriate to take it modulo 360, for example.)
-
-A special case of domain-specific validation, which Web IDL already has you
-covered for, is the 0–255 range. This can be written as <code>[EnforceRange]
-octet</code>: any JavaScript number in the range 0–255, throwing a <code
-highlight="js">TypeError</code> outside the range and rounding inside of it.
-(And indeed, if it turns out that the other power-of-two ranges are semantically
-meaningful for your domain, such that you want the modulo or range-checking
-behavior, feel free to use them.)
+  :: Any positive or zero JavaScript number between -2<sup>53</sup> and 2<sup>53</sup>,
```

Comment:
Saying positive or zero followed by -2^53 seems weird.

Reply:
Wow yeah that's just plain wrong.
```suggestion
  :: Any JavaScript number between 0 and 2<sup>54</sup>,
```